### PR TITLE
Move Service and ComponentOptions to impl.

### PR DESCRIPTION
**Pull request by RayRoestenburg**

### CODE REVIEW

File: sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/ComponentOptions.scala [hunk 1]
```diff
@@ -16,5 +16,5 @@
 
 package com.akkaserverless.scalasdk
 
-//FIXME see if needed
+//FIXME see if needed, maybe should be moved to impl
```
Comment (patriknw) [inline sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/ComponentOptions.scala:19]: if it is moved to impl we should (also) include `withForwardHeaders` and `forwardHeaders` in the more concrete options interfaces, such as EventSourcedEntityOptions

Comment (RayRoestenburg) [reply sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/ComponentOptions.scala]: yes will do

Comment (RayRoestenburg) [reply sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/ComponentOptions.scala]: These interfaces still extend ComponentOptions in impl, so should be good?

**Comment by RayRoestenburg** [comment]
Ok with this @patriknw ?

File: sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/ComponentOptions.scala [hunk 2]
```diff
@@ -14,31 +14,25 @@
...
+trait ComponentOptions {
 
   /**
-   * @return the headers requested to be forwarded as metadata (cannot be mutated, use
-   *     withForwardHeaders)
+   * @return
+   *   the headers requested to be forwarded as metadata (cannot be mutated, use withForwardHeaders)
    */
-  Set<String> forwardHeaders();
+  def forwardHeaders(): Set[String]
```
Comment (patriknw) [inline sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/ComponentOptions.scala:31]: I would be more explicit with that `Set` type. I see that there is an import `java.util.Set` above but I'd rather have the full package here, or rename the import to JSet.

File: sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/ComponentOptions.scala [hunk 3]
```diff
@@ -14,31 +14,25 @@
...
-package com.akkaserverless.javasdk;
+package com.akkaserverless.javasdk.impl
 
-import com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntityOptions;
-import com.akkaserverless.javasdk.impl.eventsourcedentity.EventSourcedEntityOptionsImpl;
+import com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntityOptions
+import com.akkaserverless.javasdk.impl.eventsourcedentity.EventSourcedEntityOptionsImpl
 
 import java.util.Collections;
 import java.util.Set;
```
Comment (patriknw) [inline sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/ComponentOptions.scala:23]: some semicolon leftovers

File: sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/ComponentOptions.scala [hunk 4]
```diff
@@ -14,31 +14,25 @@
...
+   * commands. By default no headers except "X-Server-Timing" are forwarded.
    */
-  ComponentOptions withForwardHeaders(Set<String> headers);
-
-  /** Create a default components option */
-  static ComponentOptions defaults() {
-    return new EventSourcedEntityOptionsImpl(
-        0, PassivationStrategy.defaultTimeout(), Collections.emptySet());
-  }
+  def withForwardHeaders(headers: Set[String]): ComponentOptions
```
Comment (patriknw) [inline sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/ComponentOptions.scala:37]: ^ regarding `Set`

Comment (patriknw) [reply sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/ComponentOptions.scala:37]: This is public api and it's somewhat hidden (strange) to only have it via `impl`. We should add override in all that `extends ComponentOptions`, for example ActionOptions

Comment (RayRoestenburg) [reply sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/ComponentOptions.scala:37]: ok

**Review by patriknw — APPROVED** [review]
LGTM

**Review by johanandren — APPROVED** [review]

**Pull request merged**